Barotrauma/BarotraumaShared/SharedSource/Serialization/SerializableProperty/SerializableProperty.cs

```diff
@@ -303,11 +303,47 @@ public bool TrySetValue(object parentObject, object value)
                             case "identifierarray":
                                 PropertyInfo.SetValue(parentObject, XMLExtensions.ParseIdentifierArray((string)value));
                                 return true;
+                            case "int" or "float":
+                                PropertyInfo.SetValue(parentObject, ParseNumericalOperators((string)value));
+                                return true;
                             default:
                                 DebugConsole.ThrowError($"Failed to set the value of the property \"{Name}\" of \"{parentObject}\" to {value}");
                                 DebugConsole.ThrowError($"(Cannot convert a string to a {PropertyType})");
                                 return false;
                         }
+
+                        float ParseNumericalOperators(string value)
+                        {
+                            float newVal = (float)PropertyInfo.GetValue(parentObject);
```

Comment:
This will cause a crash if the value is an integer (you cannot cast a boxed integer to a float).